### DISCUSSION TITLE
How to set the WEB UI Password correctly with PASSWORD_HASH?

### DISCUSSION THREAD

**Discussion by NetRat88** [wg-easy/wg-easy#1131] [Q&A] [answered]
Dear community,

to set the Web UI password I used the environement variable PASSWORD_HASH because the README tells me that the variable PASSWORD is deprecated.
But after deploying my docker compose no password is required to log in to the WEB UI.
If I use the variable PASSWORD it will require the password. But if I use for example the example password from the docker-compose.yml in this repo foobar123 it is not working and I get the Error "error".
How should I set the WEB UI password correctly in my docker-compose?

I am using a stack.env file with the following configuration for the passwords:

WG_PASSWORD_HASH=$$2y$$10$$hBCoykrB95WSzuV4fafBzOHWKu9sbyVa34GJr8VV5R/pIelfEMYyG

I also tried to set the HASH directly in the docker compose and used the variable PASSWORD=foobar123. Also didn't work.

```

version: "3.8"

services:
wg-easy:
env_file:
- stack.env
environment:
# ⚠️ Change the server's hostname (clients will connect to):
- WG_HOST=${WG_HOST}
# ⚠️ Change the Web UI Password:
- PASSWORD_HASH=${WG_PASSWORD_HASH}
# 💡 This is the Pi-Hole Container's IP Address
- WG_DEFAULT_DNS=10.8.1.3
- WG_DEFAULT_ADDRESS=10.8.0.x
image: ghcr.io/wg-easy/wg-easy
container_name: wireguard-easy
hostname: wireguard-docker
labels:
com.centurylinklabs.watchtower.monitor-only: "true" #nur Monitoring auf Updates durch Watchtower und kein Auto-Update
volumes:
- /home/pi/wireguard-easy:/etc/wireguard
ports:
- "51820:51820/udp"
- "51821:51821/tcp"
restart: always
cap_add:
- NET_ADMIN
- SYS_MODULE
sysctls:
- net.ipv4.ip_forward=1
- net.ipv4.conf.all.src_valid_mark=1
networks:
wg-easy:
ipv4_address: 10.8.1.2
pihole:
container_name: pihole
image: pihole/pihole:latest
hostname: pihole-docker
labels:
com.centurylinklabs.watchtower.monitor-only: "true" #nur Monitoring auf Updates durch Watchtower und kein Auto-Update
# For DHCP it is recommended to remove these ports and instead add: network_mode: "host"
ports:
- "53:53/tcp"
- "53:53/udp"
- "82:80/tcp" # Admin Web Port
env_file:
- stack.env
environment:
- WEBPASSWORD=${PIHOLE_WEBPASSWORD}
- FTLCONF_LOCAL_IPV4=${HOST_IP}
# Volumes store your data between container upgrades
volumes:
- /home/pi/pihole/etc-pihole:/etc/pihole
- /home/pi/pihole/etc-dnsmasq.d:/etc/dnsmasq.d
# https://github.com/pi-hole/docker-pi-hole#note-on-capabilities
cap_add:
- NET_ADMIN # Required if you are using Pi-hole as your DHCP server, else not needed
restart: always
networks:
wg-easy:
ipv4_address: 10.8.1.3
networks:
wg-easy:
ipam:
config:
- subnet: 10.8.1.0/24

```

**Comment by pheiduck** [comment, marked as answer]
you have to use **nightly** tag.

**Reply by mpaluch92** [reply]
IMHO this should be documented and communicated way better... 
Why does the README state that the environment variable `PASSWORD` is deprecated and replaced with `PASSWORD_HASH` if this isn't the case at all on the stable branch?

**Reply by polymorphisma** [reply]
in the docker-compose.yml file change     image: ghcr.io/wg-easy/wg-easy to image: ghcr.io/wg-easy/wg-easy:nightly. for a begineer like me who is confused where to use nightly tag.

**Reply by pheiduck** [reply]
> IMHO this should be documented and communicated way better... Why does the README state that the environment variable `PASSWORD` is deprecated and replaced with `PASSWORD_HASH` if this isn't the case at all on the stable branch?

@mpaluch92 Feel free to make a documentation PR :)

**Reply by pheiduck** [reply]
Or you guys haven't seen this: https://github.com/wg-easy/wg-easy?tab=readme-ov-file#versions ;)

**Reply by pheiduck** [reply]
maybe we should name it images.

**Reply by bckelley** [reply]
is this a freaking joke?! this doesn't work

**Comment by Bojo67** [comment]
I have the same problem!
As soon as I use a HASH password, it connects directly to the interface without asking for the password.

**Comment by mathys-lopinto** [comment]
For the bcrypt hash: you can see the documentation: [How_to_generate_an_bcrypt_hash.md](https://github.com/wg-easy/wg-easy/blob/master/How_to_generate_an_bcrypt_hash.md)

**Reply by Germain-S** [reply]
link is not working anymore :(

**Comment by 78wesley** [comment]
https://github.com/wg-easy/wg-easy/blob/production/How_to_generate_an_bcrypt_hash.md
